content/lessons/chapter-6/in-out-4.tsx

```diff
@@ -1,34 +1,221 @@
 'use client'
 
+import { ScriptingChallenge, LessonInfo, CodeExample, Title, Table } from 'ui'
+import { EditorConfig } from 'types'
 import { useTranslations } from 'hooks'
-import { Introduction, Text } from 'ui'
+import { Text } from 'ui'
+import { useState } from 'react'
+import { getLessonKey } from 'lib/progress'
+import { useAuthContext } from 'providers/AuthProvider'
 
 export const metadata = {
-  title: 'chapter_six.in_out_one.title',
-  image: '/assets/images/chapter-4-intro-1.jpg',
+  title: 'chapter_six.in_out_four.title',
   key: 'CH6INO4',
 }
 
 export default function InOut4({ lang }) {
   const t = useTranslations(lang)
+  const tableHeading = [
+    t('chapter_six.in_out_four.table.heading.one'),
+    t('chapter_six.in_out_four.table.heading.two'),
+    t('chapter_six.in_out_four.table.heading.three'),
+    t('chapter_six.in_out_four.table.heading.four'),
+  ]
+  const outputRows = [
+    [
+      t('chapter_six.in_out_four.table.row_one.column.one'),
+      t('chapter_six.in_out_four.table.row_one.column.two'),
+      t('chapter_six.in_out_four.table.row_one.column.three'),
+      t('chapter_six.in_out_four.table.row_one.column.four'),
+    ],
+    [
+      t('chapter_six.in_out_four.table.row_two.column.one'),
+      t('chapter_six.in_out_four.table.row_two.column.two'),
+      t('chapter_six.in_out_four.table.row_two.column.three'),
+      t('chapter_six.in_out_four.table.row_two.column.four'),
+    ],
+    [
+      t('chapter_six.in_out_four.table.row_three.column.one'),
+      t('chapter_six.in_out_four.table.row_three.column.two'),
+      t('chapter_six.in_out_four.table.row_three.column.three'),
+      t('chapter_six.in_out_four.table.row_three.column.four'),
+    ],
+    [
+      t('chapter_six.in_out_four.table.row_four.column.one'),
+      t('chapter_six.in_out_four.table.row_four.column.two'),
+      t('chapter_six.in_out_four.table.row_four.column.three'),
+      t('chapter_six.in_out_four.table.row_four.column.four'),
+    ],
+    [
+      t('chapter_six.in_out_four.table.row_five.column.one'),
+      t('chapter_six.in_out_four.table.row_five.column.two'),
+      t('chapter_six.in_out_four.table.row_five.column.three'),
+      t('chapter_six.in_out_four.table.row_five.column.four'),
+    ],
+  ]
+
+  const { account } = useAuthContext()
+
+  const javascript = {
+    program: `
+const testAddress = 'bc1p5d7rjq7g6rdk2yhzks9smlaqtedr4dekq08ge8ztwac72sfr9rusxg3297'
+const outputInstance = Output.from_options(testAddress, 1000000);
+console.log(outputInstance.serialize().toString('hex')==='40420f0000000000220120a37c3903c8d0db6512e2b40b0dffa05e5a3ab73603ce8c9c4b7771e5412328f9' && 'true')
```

Comment:
Lets change the validation to use Matthew's challenge code for consistency

https://github.com/saving-satoshi/challenges/blob/master/chapter6/javascript/challenge.js#L66

otherwise looking good! almost there!